zokrates_github_resolver/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+//! ```
+//!
+//! For example:
+//! ```zokrates
+//! import "github.com/Zokrates/ZoKrates/master/zokrates_cli/examples/merkleTree/sha256PathProof3.code" as merkleTreeProof
+//! ```
+//!
+//! Example above imports file `zokrates_cli/examples/merkleTree/sha256PathProof3.code` located at ZoKrates
+//! repository's `master` branch by downloading from URL:
+//! https://raw.githubusercontent.com/Zokrates/ZoKrates/master/zokrates_cli/examples/merkleTree/sha256PathProof3.code
+//!
+
+use reqwest;
+use std::fs::File;
+use std::io::{self, copy, BufReader};
+use std::path::PathBuf;
+use tempfile::NamedTempFile;
+
+use mockito;
+use mockito::Mock;
+
+/// Prefix for github import source to be distinguished.
+const GITHUB_IMPORT_PREFIX: &str = "github.com/";
+
+/// GitHub download URL base
+const GITHUB_URL_BASE: &str = "https://raw.githubusercontent.com";
+
+/// Resolves import from the Github.
+/// This importer needs to be provided with location since relative paths could be used inside the
+/// files that are imported from github.
+pub fn resolve(
+    location: &Option<String>,
+    path: &String,
+) -> Result<(BufReader<File>, String, String), io::Error> {
+    if let Some(location) = location {
+        let (root, repo, branch, path) = parse_input_path(&path)?;
+
+        let pb = download_from_github(GITHUB_URL_BASE, &root, &repo, &branch, &path)?;
+        let file = File::open(&pb)?;
+        let br = BufReader::new(file);
+
+        let alias = PathBuf::from(path.clone())
+            .as_path()
+            .file_stem()
+            .unwrap()
+            .to_string_lossy()
+            .to_string();
+
+        Ok((br, location.to_owned(), alias))
+    } else {
+        Err(io::Error::new(io::ErrorKind::Other, "No location provided"))
+    }
+}
+
+/// Similar to the `resolve`, but uses mock for github for `examples/import_github.code` test and
+/// not require internet connection.
+pub fn resolve_with_mock(
+    location: &Option<String>,
+    path: &String,
+) -> Result<(BufReader<File>, String, String), io::Error> {
+    if let Some(location) = location {
+        let (_m1, _m2) = init_github_mock(&location);
+
+        let (root, repo, branch, path) = parse_input_path(&path)?;
+
+        let pb =
+            download_from_github(mockito::server_url().as_str(), &root, &repo, &branch, &path)?;
+        let file = File::open(&pb)?;
+        let br = BufReader::new(file);
+
+        let alias = PathBuf::from(path.clone())
+            .as_path()
+            .file_stem()
+            .unwrap()
+            .to_string_lossy()
+            .to_string();
+
+        Ok((br, location.to_owned(), alias))
+    } else {
+        Err(io::Error::new(io::ErrorKind::Other, "No location provided"))
+    }
+}
+
+/// Initializes github.com mocks for `import_github` example to run from `examples` tests.
+/// Note that returned mock objects should be alive prior to github requests.
+fn init_github_mock(location: &str) -> (Mock, Mock) {
```

Comment:
I tried it first, but then I figured out that it only works for tests that are running in `zokrates_github_resolver` crate. But when you run tests for `zokrates_cli` crate, said resolver crate is built without `#[cfg(test)]` flag. You can try it yourself. That’s the reason why I omitted this flag from definitions and use mockito crate in non-dev dependencies. Don’t know how to circumvent this issue though.

Reply:
I had a thought about it. I think mocks should be generated by each test. So if we were testing anything outside `zokrates_github_resolver`, we would have to pass a mock server to the resolver. For now, I reckon we should test only in `zokrates_github_resolver`. I made a PR on your branch with an idea of what it could look like.